core/src/test/java/org/elasticsearch/test/rest/client/http/HttpRequestBuilder.java

```diff
@@ -86,14 +86,42 @@ public HttpRequestBuilder port(int port) {
         return this;
     }
 
+    /**
+     * Sets the path to send the request to. Url encoding needs to be applied by the caller.
+     * Use {@link #pathParts(String...)} instead if the path needs to be encoded, part by part.
+     */
     public HttpRequestBuilder path(String path) {
         this.path = path;
         return this;
     }
 
+    /**
+     * Sets the path by providing the different parts (without slashes), which will be properly encoded.
+     */
+    public HttpRequestBuilder pathParts(String... path) {
+        //encode rules for path and query string parameters are different. We use URI to encode the path, and URLEncoder for each query string parameter (see addParam).
+        //We need to encode each path part separately though, as each one might contain slashes that need to be escaped, which needs to be done manually.
+        if (path.length == 0) {
+            this.path = "/";
+            return this;
+        }
+        StringBuilder finalPath = new StringBuilder();
+        for (String pathPart : path) {
+            try {
+                finalPath.append('/');
+                URI uri = new URI(null, null, null, -1, pathPart, null, null);
+                //manually escape any slash that each part may contain
+                finalPath.append(uri.getRawPath().replaceAll("/", "%2F"));
```

Comment:
URLEncoder is good only for query_string params, not for url path. They have different escaping rules according to the spec. Also URI applies the right rules to the path, but given that you have to provide the whole path it won't escape slashes. This is frustrating...but I couldn't find any better solutions.